packages/react-components/react-table/src/hooks/useSelection.ts

```diff
@@ -1,14 +1,38 @@
-import { GetRowIdInternal, SelectionStateInternal, UseTableOptions } from './types';
-import { useMultipleSelection } from './useMultipleSelection';
-import { useSingleSelection } from './useSingleSelection';
+import * as React from 'react';
+import { useEventCallback, usePrevious } from '@fluentui/react-utilities';
+import { createSelectionManager } from './selectionManager';
+import { GetRowIdInternal, RowId, SelectionMode, SelectionStateInternal } from './types';
 
 export function useSelection<TItem>(
-  selectionMode: UseTableOptions<TItem>['selectionMode'],
+  selectionMode: SelectionMode,
   items: TItem[],
   getRowId: GetRowIdInternal<TItem>,
 ): SelectionStateInternal {
-  const multipleSelectionState = useMultipleSelection(items, getRowId);
-  const singleSelectionState = useSingleSelection();
+  const prevSelectionMode = usePrevious(selectionMode);
+  const [selected, setSelected] = React.useState(() => new Set<RowId>());
```

Comment:
I see that selection managers are using `SelectionItemId` instead of `RowId`. Is it intentional separation?

Reply:
yeah, it's intentional. This manager is just to manage some kind of keyed items